WooCommerce/Classes/ViewModels/Booking Details/AppointmentDetailsContent.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+import struct Networking.Booking
+
+extension BookingDetailsViewModel {
+    struct AppointmentDetailsContent {
+        static let appointmentDateFormatter = {
+            let dateFormatter = DateFormatter()
+            dateFormatter.dateFormat = "EEEE, dd MMMM yyyy"
+            return dateFormatter
+        }()
+
+        static let appointmentTimeFrameFormatter = {
+            let dateFormatter = DateFormatter()
+            dateFormatter.dateFormat = "hh:mm a"
+            return dateFormatter
+        }()
+
+        struct Row: Identifiable {
+            let title: String
+            let value: String
+
+            var id: String {
+                return title
+            }
+        }
+
+        let rows: [Row]
+
+        init(_ booking: Booking) {
+            let durationMinutes = Int(booking.endDate.timeIntervalSince(booking.startDate) / 60)
+            let appointmentDate = Self.appointmentDateFormatter.string(from: booking.startDate)
+            let appointmentTimeFrame = [
+                Self.appointmentTimeFrameFormatter.string(from: booking.startDate),
+                Self.appointmentTimeFrameFormatter.string(from: booking.endDate)
```

Comment:
Nit: these can be simplified:

```suggestion
            let appointmentDate = booking.startDate.formatted(date: .numeric, time: .omitted)
            let appointmentTimeFrame = [
                booking.startDate.formatted(date: .omitted, time: .shortened),
                booking.endDate.formatted(date: .omitted, time: .shortened)
```

Reply:
Done in [1220ac5](https://github.com/woocommerce/woocommerce-ios/pull/16168/commits/1220ac5eb1f1a4b3b10cb5df308634bc70e10753)